tests/Microsoft.Data.Analysis.Tests/DataFrameTests.cs

```diff
@@ -1947,84 +1947,170 @@ public void TestMutationOnRows()
             }
         }
 
+        [Fact]
+        public void TestAppendRows()
+        {
+            DataFrame df = MakeDataFrame<float, bool>(10);
+            DataFrame df2 = MakeDataFrame<int, bool>(5);
+            Assert.Equal(10, df.Rows.Count);
+            Assert.Equal(1, df.Columns[0].NullCount);
+            Assert.Equal(1, df.Columns[1].NullCount);
+
+            DataFrame ret = df.Append(df2.Rows, inPlace: false);
+            Assert.Equal(10, df.Rows.Count);
+            Assert.Equal(1, df.Columns[0].NullCount);
+            Assert.Equal(1, df.Columns[1].NullCount);
+
+            Verify(ret, df, df2);
+
+            void Verify(DataFrame ret, DataFrame check1, DataFrame check2)
+            {
+                Assert.Equal(15, ret.Rows.Count);
+                Assert.Equal(2, ret.Columns[0].NullCount);
+                Assert.Equal(2, ret.Columns[1].NullCount);
+                for (long i = 0; i < ret.Rows.Count; i++)
+                {
+                    DataFrameRow row = ret.Rows[i];
+                    for (int j = 0; j < check1.Columns.Count; j++)
+                    {
+                        if (i < check1.Rows.Count)
+                        {
+                            Assert.Equal(row[j], check1.Rows[i][j]);
+                        }
+                        else
+                        {
+                            Assert.Equal(row[j]?.ToString(), (check2.Rows[i - check1.Rows.Count][j])?.ToString());
```

Comment:
Why are you `ToString()`'ing here, but not in the `Assert.Equal` above that is checking `check1`?

Reply:
Because the types don't match(but the values do) as part of testing `Append`. `df.Columns[0]` is `float` while `df2.Columns[1]` is `int`. So the assert would complain that I'm comparing a float(from `row`) and an int(from `check2`). `Check1.Columns[0]` holds `float`s, so the types match and the Assert will succeed. Hence the `ToString`.